modules/core_project_factory/main.tf

```diff
@@ -26,18 +26,17 @@ resource "random_id" "random_project_id_suffix" {
  *****************************************/
 locals {
   group_id               = "${var.manage_group ? format("group:%s", var.group_email) : ""}"
-  project_id             = "${google_project.main.project_id}"
-  project_number         = "${google_project.main.number}"
   project_org_id         = "${var.folder_id != "" ? "" : var.org_id}"
   project_folder_id      = "${var.folder_id != "" ? var.folder_id : ""}"
-  temp_project_id        = "${var.random_project_id ? format("%s-%s",var.name,random_id.random_project_id_suffix.hex) : var.name}"
+  project_id             = "${var.project_id == "" ? var.name : var.project_id}"
+  temp_project_id        = "${var.random_project_id ? format("%s-%s",local.project_id,random_id.random_project_id_suffix.hex) : local.project_id}"
   s_account_fmt          = "${format("serviceAccount:%s", google_service_account.default_service_account.email)}"
-  api_s_account          = "${format("%s@cloudservices.gserviceaccount.com", local.project_number)}"
+  api_s_account          = "${format("%s@cloudservices.gserviceaccount.com", google_project.main.number)}"
```

Comment:
Why can't this continue to reference `project_number`?

Reply:
The net effect of this change is a no-op. It removes a level of indirection since `local.project_number` is referring to `google_project.main.number`. I see no purpose for continuing to use `local.project_number`. It makes things a bit more difficult to read so I would much prefer to remove it.